Sources/Testing/Expectations/Expectation.swift

```diff
@@ -41,6 +39,38 @@ public struct Expectation: Sendable {
 
   /// The source location where this expectation was evaluated.
   public var sourceLocation: SourceLocation
+
+  ///
+  public init(
+    mismatchedErrorDescription: String? = nil,
+    differenceDescription: String? = nil,
+    isPassing: Bool,
+    isRequired: Bool,
+    sourceLocation: SourceLocation
+  ) {
+    self.evaluatedExpression = Expression("")
+    self.mismatchedErrorDescription = mismatchedErrorDescription
+    self.differenceDescription = differenceDescription
+    self.isPassing = isPassing
+    self.isRequired = isRequired
+    self.sourceLocation = sourceLocation
+  }
```

Comment:
I'd like to explore ideas for ways to structure this that avoid needing to publicly expose an `init` on `Expectation`. I know I brought that idea up in an earlier comment on this PR, but after reviewing this type again, and taking @grynspan 's comments about the two properties above into consideration, I'm wary to open this type up for general API use right now until it's a bit better designed.

As one alternative, I wonder if we could expose a `static func` on `Issue.Kind` which takes a generic message, as well as one or more values of the evaluated expression, and internally construct an `Expectation` with the relevant `Expression` and `Expression.Value` values representing all of the passed-in values? Something like

```swift
extension Issue.Kind {
  public static func expectationFailed<each T>(
    _ description: String,
    values: repeat each T,
    isRequired: Bool,
    sourceLocation: SourceLocation
  ) -> Self
}
```

This would require some internal change to `__Expression` but that seems doable, and it would allow capturing rich details of the left- and right-hand side values of binary expressions.